types/backbone/backbone-tests.ts

```diff
@@ -564,6 +564,38 @@ class ModellessView extends Backbone.View {
     }
 }
 
+class EventsViewObject extends Backbone.View {
+    events: Backbone.EventsHash;
+
+    constructor(options: Backbone.ViewOptions) {
+        super(options);
+
+        this.events = {
+            "click": "onClick",
+        };
+
+        if (typeof options.events === "object") {
+            this.events = options.events;
+        }
+    }
+}
+
+class EventsViewMethod extends Backbone.View {
+    events: () => Backbone.EventsHash;
+
+    constructor(options: Backbone.ViewOptions) {
+        super(options);
+
+        this.events = () => ({
+            "click": "onClick",
+        });
```

Comment:
Could you do this assignment with an old-fashioned `function` that tries to access `this`? I'm asking because I think accessing `this` is the main reason why someone might do a method instead of a static `events` hash.

Reply:
With `noImplicitThis` set to `true` it has correct typing, otherwise, `this` is `any` inside function declarations.

Reply:
I think you mean to imply something but I'm not following. Could you elaborate?

Reply:
Did you mean this?

```js
this.events = function() {
    this; // Should have correct typing, not `any`

    return {
        "click": "onClick",
    }
}
```

Reply:
Maybe something more like this:

```ts
function events(this: EventsViewObject) {
    const eventsHash = {
        click: 'onClick',
    };
    if (this.model) eventsHash.reset = 'onReset';
    return eventsHash;
}

class EventsViewObject extends View {
    constructor(options: ViewOptions) {
        super(options);
        this.events = events;
    }
}
```

Do the tests run with or without `noImplicitThis`?

Reply:
Sure, if you explicitly type `this` in that way it works fine. Would you like me to add that to the tests?

The tests fail with `"noImplicitThis": true`, but from errors where we're passing `this` to `.once` methods:

```
/DefinitelyTyped/types/backbone/backbone-tests.ts
   40:40  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6 compile error: 
'this' implicitly has type 'any' because it does not have a type annotation  @definitelytyped/expect
   42:61  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6 compile error: 
'this' implicitly has type 'any' because it does not have a type annotation  @definitelytyped/expect
   48:30  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6 compile error: 
'this' implicitly has type 'any' because it does not have a type annotation  @definitelytyped/expect
  372:45  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6 compile error: 
'this' implicitly has type 'any' because it does not have a type annotation  @definitelytyped/expect
  374:66  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6 compile error: 
'this' implicitly has type 'any' because it does not have a type annotation  @definitelytyped/expect
```

Reply:
Yes, please add it to the tests.

Reply:
I've added the assignment in `EventsViewMethod`